src/topology/separation.lean

```diff
@@ -16,6 +16,46 @@ variables {α : Type u} {β : Type v} [topological_space α]
 
 section separation
 
+/--
+`separate` is a predicate on pairs of `finset`s of a topological space.  It holds if the two
+`finset`s are contained in disjoint open sets.
+-/
+def separate : finset α → finset α → Prop :=
+  λ (s t : finset α), disjoint s t → ∃ U V : (set α), (is_open U) ∧ is_open V ∧
+  (∀ a : α, a ∈ s → a ∈ U) ∧ (∀ a : α, a ∈ t → a ∈ V) ∧ disjoint U V
+
+lemma separate_symm (s t : finset α) : separate s t → separate t s :=
+begin
+ intros h1 d,
+ obtain ⟨U, V, oU, oV, aU, bV, UV⟩ := h1 (disjoint.symm d),
+ exact ⟨V, U, oV, oU, bV, aU, disjoint.symm UV⟩
+end
+
+lemma separate_empty_right : ∀ {a : finset α}, separate a ∅ :=
+λ a d, ⟨_, _, is_open_univ, is_open_empty, λ a h, mem_univ a, λ a h, by cases h, disjoint_empty _⟩
+
+lemma separate_union_of : ∀ {a b c : finset α}, separate a c → separate b c → separate (a ∪ b) c :=
+begin
+  intros a b c ac bc d,
+  obtain ⟨U, V, oU, oV, aU, bV, UV⟩ :=
+    ac (finset.disjoint_of_subset_left (finset.subset_union_left _ _) d),
+  obtain ⟨W, X, oW, oX, aW, bX, WX⟩ :=
+    bc (finset.disjoint_of_subset_left (finset.subset_union_right _ _) d),
+  refine ⟨U ∪ W, V ∩ X, is_open_union oU oW, is_open_inter oV oX,
+    λ x xab, _, λ x xc, ⟨bV _ xc, bX _ xc⟩, _⟩,
+  { cases finset.mem_union.mp xab with h h,
+    { exact mem_union_left W (aU x h) },
+    { exact mem_union_right U (aW x h) } },
+  { apply set.disjoint_union_left.mpr,
+    exact ⟨disjoint_of_subset_right (inter_subset_left _ _) UV,
+      disjoint_of_subset_right (inter_subset_right _ _) WX⟩ },
+end
+
+lemma separate_of_singletons :
+  (∀ a b, separate ({a} : finset α) {b}) → (∀ s t, separate (s : finset α) t) :=
```

Comment:
You can move the assumption left of the `:`. Then you don't need to start with a lambda.